onyxia-api/src/main/java/fr/insee/onyxia/api/openapi/OpenApiConfiguration.java

```diff
@@ -1,12 +1,72 @@
 package fr.insee.onyxia.api.openapi;
 
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.security.OAuthFlow;
+import io.swagger.v3.oas.models.security.OAuthFlows;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springdoc.core.customizers.OperationCustomizer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
-import io.swagger.v3.oas.annotations.security.SecurityScheme;
 
 @Configuration
-@SecurityScheme(type = SecuritySchemeType.HTTP, scheme = "bearer", bearerFormat = "JWT",name = "auth")
 public class OpenApiConfiguration {
 
+
+    private static final Logger logger = LoggerFactory.getLogger(OpenApiConfiguration.class);
+
+    @Value("${keycloak.auth-server-url}")
+    public String keycloakUrl;
+
+    @Value("${keycloak.realm}")
+    public String realmName;
+
+    public final String SCHEMEKEYCLOAK = "oAuthScheme";
+
+    @Bean
+    @ConditionalOnProperty(name = "com.example.demo.auth.type", havingValue = "openidconnect", matchIfMissing = true)

```

Comment:
I don't think we want `matchIfMissing=true` here. 
Also, the name is wrong. Our auth type property key is `authentication.mode`.